src/services/member/plugins/action/service.ts

```diff
@@ -1,15 +1,54 @@
-import { CannotModifyOtherMembers } from '../../../../utils/errors';
+import { ActionTriggers } from '@graasp/sdk';
+
+import { CannotModifyOtherMembers, UnauthorizedMember } from '../../../../utils/errors';
 import { Repositories } from '../../../../utils/repositories';
 import { ActionService } from '../../../action/services/action';
+import { ItemActionType } from '../../../item/plugins/action/utils';
 import { Actor } from '../../entities/member';
 
+export const getMonthBeforeNow = () => {
+  const date = new Date(); // Today's date
+  date.setMonth(date.getMonth() - 1); // Set the date to one month ago
+  return date;
+};
+
 export class ActionMemberService {
   actionService: ActionService;
 
   constructor(actionService: ActionService) {
     this.actionService = actionService;
   }
 
+  async getFilteredActions(
+    actor: Actor,
+    repositories: Repositories,
+    filters: { startDate?: string; endDate?: string },
+  ) {
+    const { actionRepository } = repositories;
+
+    const { startDate, endDate } = filters;
+    const start = startDate ? new Date(startDate) : getMonthBeforeNow();
+    const end = endDate ? new Date(endDate) : new Date();
+
+    const allowedTypes = [
+      ActionTriggers.MemberLogin,
+      ItemActionType.Create,
+      ItemActionType.Update,
+      ItemActionType.Delete,
+    ];
```

Comment:
Why not get everything, or a subset given by the user? It feels a bit random which types you're fetching.